src/corePlugins.js

```diff
@@ -409,8 +409,8 @@ export let variantPlugins = {
     matchVariant(
       'supports',
       (value = '') => {
-        let check = normalize(value)
-        let isRaw = /^\w*\s*\(/.test(check)
+        let check = value.startsWith('--') ? value : normalize(value)
+        let isRaw = /^[\w-]*\s*\(/.test(check)
```

Comment:
2 changes made
- added a `value.startsWith('--')` pre-normalization check that operates in line with #13537 to prevent automatic `var(…)` injection when checking support for custom properties
   
   example: `supports-[--test]:flex`:
   - ❌`@supports var(--test)`
   - ✅`@supports (--test: var(--tw))`
- added hyphen-support for regex pattern matching raw `supports` queries